handlers/misttriggers/push_end.go

```diff
@@ -32,20 +35,39 @@ func (d *MistCallbackHandlersCollection) TriggerPushEnd(w http.ResponseWriter, r
 	streamName := lines[1]
 	// TODO: Left commented as these will all be used by the next piece we'll pull out of https://github.com/livepeer/catalyst-api/pull/30
 	// destination := lines[2]
-	// actualDestination := lines[3]
-	// pushStatus := lines[5]
+	actualDestination := lines[3]
+	pushStatus := lines[5]
 
 	switch streamNameToPipeline(streamName) {
 	case Transcoding:
 		// TODO: Left commented for illustration of the alternate code path here as this is the next piece we'll pull out of https://github.com/livepeer/catalyst-api/pull/30
 		// d.TranscodingPushEnd(w, req, streamName, destination, actualDestination, pushStatus)
 	case Segmenting:
 		d.SegmentingPushEnd(w, req, streamName)
+	case Recording:
+		d.RecordingPushEnd(w, req, streamName, actualDestination, pushStatus)
 	default:
 		// Not related to API logic
 	}
 }
 
+func (d *MistCallbackHandlersCollection) RecordingPushEnd(w http.ResponseWriter, req *http.Request, streamName, actualDestination, pushStatus string) {
+	var err error
+	pushSuccess := pushStatus == "null"
```

Comment:
What is the expected behavior from Studio if this is `false`? Should it show an error to the user somehow or can it still recover the recording somehow?

Reply:
Fail in our first version. Improve by recovering in later tasks. Same applies for transcoding stages where some of them can fail independently. To be optimal we can just re-run failed stages. For now we re-run entire process.